eng/packages/http-client-csharp/generator/Azure.Generator/src/Utilities/ResourceDetection.cs

```diff
@@ -8,16 +8,49 @@
 using System.Collections.Concurrent;
 using System.Diagnostics.CodeAnalysis;
 using System.Linq;
+using System.Text;
 
 namespace Azure.Generator.Utilities
 {
     internal class ResourceDetection
     {
         private const string ProvidersSegment = "/providers/";
-        private ConcurrentDictionary<string, (string Name, InputModelType? InputModel)?> _resourceDataSchemaCache = new ConcurrentDictionary<string, (string Name, InputModelType? InputModel)?>();
+        private const string ResourceGroupScopePrefix = "/subscriptions/{subscriptionId}/resourceGroups";
+        private const string SubscriptionScopePrefix = "/subscriptions";
+        private const string TenantScopePrefix = "/tenants";
 
-        private static InputModelType? FindObjectSchemaWithName(string name)
-            => AzureClientPlugin.Instance.InputLibrary.InputNamespace.Models.OfType<InputModelType>().FirstOrDefault(inputModel => inputModel.Name == name);
+        private ConcurrentDictionary<RequestPath, (string Name, InputModelType? InputModel)?> _resourceDataSchemaCache = new ConcurrentDictionary<RequestPath, (string Name, InputModelType? InputModel)?>();
+
+        public bool IsResource(OperationSet set) => TryGetResourceDataSchema(set, out _, out _);
+
+        public static string GetResourceTypeFromPath(RequestPath requestPath)
+        {
+            var index = ((string)requestPath).LastIndexOf(ProvidersSegment);
```

Comment:
these casts read quite weird.
Could we have a local variable such as 
```
string rawRequestPath = requestPath.ToString();
```
?

Reply:
but if we are checking the index of provider segment, could we do it by checking the segments in `RequestPath` instance?
Maybe `RequestPath` could have such a property for the index of `providers` segment itself since it is so widely used here.